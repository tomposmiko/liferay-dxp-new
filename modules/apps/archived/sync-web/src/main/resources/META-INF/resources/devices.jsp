<%--
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */
--%>

<%@ include file="/init.jsp" %>

<%
String tabs1 = ParamUtil.getString(request, "tabs1", "devices");

String orderByCol = ParamUtil.getString(request, "orderByCol", "name");
String orderByType = ParamUtil.getString(request, "orderByType", "asc");

PortletURL portletURL = PortletURLBuilder.createRenderURL(
	renderResponse
).setTabs1(
	tabs1
).buildPortletURL();

SearchContainer<SyncDevice> syncDeviceSearchContainer = new SearchContainer<>(renderRequest, portletURL, null, "no-devices-were-found");

syncDeviceSearchContainer.setOrderByCol(orderByCol);

OrderByComparator<SyncDevice> orderByComparator = null;

if (orderByCol.equals("name")) {
	orderByComparator = OrderByComparatorFactoryUtil.create("SyncDevice", "userName", orderByType.equals("asc"));
}
else if (orderByCol.equals("build")) {
	orderByComparator = OrderByComparatorFactoryUtil.create("SyncDevice", "buildNumber", orderByType.equals("asc"));
}
else if (orderByCol.equals("last-seen")) {
	orderByComparator = OrderByComparatorFactoryUtil.create("SyncDevice", "modifiedDate", orderByType.equals("asc"));
}
else {
	orderByComparator = OrderByComparatorFactoryUtil.create("SyncDevice", orderByCol, orderByType.equals("asc"));
}

syncDeviceSearchContainer.setOrderByComparator(orderByComparator);

syncDeviceSearchContainer.setOrderByType(orderByType);

String portletId = (String)request.getAttribute(WebKeys.PORTLET_ID);

if (portletId.equals(SyncPortletKeys.SYNC_ADMIN_PORTLET)) {
	String keywords = ParamUtil.getString(request, "keywords");

	syncDeviceSearchContainer.setResultsAndTotal(SyncDeviceLocalServiceUtil.search(themeDisplay.getCompanyId(), keywords, syncDeviceSearchContainer.getStart(), syncDeviceSearchContainer.getEnd(), orderByComparator));
}
else {
	syncDeviceSearchContainer.setResultsAndTotal(SyncDeviceLocalServiceUtil.getSyncDevices(themeDisplay.getUserId(), syncDeviceSearchContainer.getStart(), syncDeviceSearchContainer.getEnd(), orderByComparator));
}
%>

<clay:management-toolbar
	managementToolbarDisplayContext="<%= new DevicesManagementToolbarDisplayContext(request, liferayPortletRequest, liferayPortletResponse, syncDeviceSearchContainer) %>"
/>

<clay:container-fluid>
	<aui:form method="post" name="fm">
		<aui:input name="redirect" type="hidden" value="<%= currentURL %>" />

		<liferay-ui:search-container
			searchContainer="<%= syncDeviceSearchContainer %>"
		>
			<liferay-ui:search-container-row
				className="com.liferay.sync.model.SyncDevice"
				escapedModel="<%= true %>"
				keyProperty="syncDeviceId"
				modelVar="syncDevice"
			>
				<liferay-ui:search-container-column-text
					cssClass="content-column name-column title-column"
					name="name"
					property="userName"
				/>

				<liferay-ui:search-container-column-text
					name="location"
					value="<%= syncDevice.getHostname() %>"
				/>

				<liferay-ui:search-container-column-text
					name="type"
					orderable="<%= true %>"
				/>

				<liferay-ui:search-container-column-text
					name="build"
					property="buildNumber"
				/>

				<liferay-ui:search-container-column-date
					name="last-seen"
					property="modifiedDate"
				/>

				<liferay-ui:search-container-column-text
					name="status"
					translate="<%= true %>"
					value="<%= SyncDeviceConstants.getStatusLabel(syncDevice.getStatus()) %>"
				/>

				<liferay-ui:search-container-column-jsp
					align="right"
					cssClass="entry-action-column"
					path="/devices_action.jsp"
				/>
			</liferay-ui:search-container-row>

			<liferay-ui:search-iterator
				markupView="lexicon"
			/>
		</liferay-ui:search-container>
	</aui:form>
</clay:container-fluid>