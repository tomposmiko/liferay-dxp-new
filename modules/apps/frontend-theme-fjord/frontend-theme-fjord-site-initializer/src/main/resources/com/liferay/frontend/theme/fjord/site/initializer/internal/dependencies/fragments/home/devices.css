.fjord-devices-image {
	background-image: url("[resources:background_2.jpg]");
	bottom: 0;
	right: 0;
	top: 0;
}