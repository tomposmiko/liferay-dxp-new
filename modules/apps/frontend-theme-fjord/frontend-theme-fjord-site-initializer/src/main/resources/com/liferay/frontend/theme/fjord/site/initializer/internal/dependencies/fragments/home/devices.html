<div class="jumbotron tertiary-background-image">
	<section class="container-fluid container-fluid-max-xl">
		<div class="row">
			<div class="col-lg-5 col-md-6 col-sm-8">
				<h1 class="display-4">
					<lfr-editable id="title" type="rich-text">The Fitness App Thatâ€™s Ready When You Are</lfr-editable>
				</h1>

				<lfr-editable id="description" type="rich-text">
					<p class="lead">Add a short explanation of how to use your app. Link to demos, samples, or forums so your customers can learn more.</p>
					<h3>30-Day Guarantee</h3>
					<p>Include reminders on discounts, promotions, or special pricing to seal the deal.</p>
					<a class="btn btn-primary btn-theme-info btn-xl" href="#">Get Started</a>
				</lfr-editable>
			</div>

			<div class="col-lg-6 col-md-6 d-none d-sm-block" style="background-image: url($background_2.jpg$);">&nbsp;</div>
		</div>
	</section>
</div>