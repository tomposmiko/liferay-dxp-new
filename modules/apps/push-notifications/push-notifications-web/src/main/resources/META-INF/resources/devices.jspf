<%--
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */
--%>

<%
String orderByCol = ParamUtil.getString(request, "orderByCol", "platform");
String orderByType = ParamUtil.getString(request, "orderByType", "asc");

SearchContainer<PushNotificationsDevice> pushNotificationsDeviceSearchContainer = new SearchContainer<>(renderRequest, portletURL, null, "there-are-no-devices");

pushNotificationsDeviceSearchContainer.setOrderByCol(orderByCol);
pushNotificationsDeviceSearchContainer.setOrderByComparator(PushNotificationsUtil.getPushNotificationsDeviceOrderByComparator(orderByCol, orderByType));
pushNotificationsDeviceSearchContainer.setOrderByType(orderByType);

pushNotificationsDeviceSearchContainer.setResultsAndTotal(() -> PushNotificationsDeviceLocalServiceUtil.getPushNotificationsDevices(pushNotificationsDeviceSearchContainer.getStart(), pushNotificationsDeviceSearchContainer.getEnd(), pushNotificationsDeviceSearchContainer.getOrderByComparator()), PushNotificationsDeviceLocalServiceUtil.getPushNotificationsDevicesCount());

ResourceBundleLoaderProvider resourceBundleLoaderProvider = (ResourceBundleLoaderProvider)request.getAttribute(PushNotificationsWebKeys.RESOURCE_BUNDLE_LOADER_PROVIDER);
%>

<clay:management-toolbar
	managementToolbarDisplayContext="<%= new PushNotificationsManagementToolbarDisplayContext(request, liferayPortletRequest, liferayPortletResponse, pushNotificationsDeviceSearchContainer) %>"
/>

<clay:container-fluid>
	<liferay-ui:search-container
		searchContainer="<%= pushNotificationsDeviceSearchContainer %>"
	>
		<liferay-ui:search-container-row
			className="com.liferay.push.notifications.model.PushNotificationsDevice"
			escapedModel="<%= true %>"
			keyProperty="pushNotificationsDeviceId"
			modelVar="pushNotificationsDevice"
		>

			<%
			User deviceUser = UserLocalServiceUtil.getUser(pushNotificationsDevice.getUserId());
			%>

			<liferay-ui:search-container-column-text
				cssClass="table-cell-expand table-title"
				name="full-name"
				value="<%= deviceUser.getFullName() %>"
			/>

			<liferay-ui:search-container-column-text
				cssClass="table-cell-expand"
				property="token"
			/>

			<%
			ResourceBundleLoader platformResourceBundleLoader = resourceBundleLoaderProvider.getResourceBundleLoader(pushNotificationsDevice.getPlatform());

			ResourceBundle platformResourceBundle = platformResourceBundleLoader.loadResourceBundle(PortalUtil.getLocale(request));
			%>

			<liferay-ui:search-container-column-text
				cssClass="table-cell-expand"
				name="platform"
				value="<%= LanguageUtil.get(platformResourceBundle, pushNotificationsDevice.getPlatform()) %>"
			/>

			<liferay-ui:search-container-column-jsp
				path="/push_notifications_device_action.jsp"
			/>
		</liferay-ui:search-container-row>

		<liferay-ui:search-iterator
			markupView="lexicon"
		/>
	</liferay-ui:search-container>
</clay:container-fluid>