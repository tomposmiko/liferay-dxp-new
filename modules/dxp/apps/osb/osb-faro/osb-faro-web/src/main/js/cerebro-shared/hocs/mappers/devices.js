import {getDeviceLabel} from 'shared/util/lang';
import {getPercentage} from 'shared/util/util';
import {getVariables, safeResultToProps} from 'shared/util/mappers';
import {sortBy} from 'lodash';

const MAX_BROWSER = 8;
const MAX_SYSTEMS = 3;

const groupBrowserData = (data, max) => {
	const orderedData = sortBy(data, 'value').reverse();

	if (orderedData.length <= max) {
		return orderedData;
	}

	const totalOtherValue = orderedData
		.slice(max)
		.reduce((actual, next) => actual + next.value, 0);

	return [
		...orderedData.slice(0, max),
		{
			value: totalOtherValue,
			valueKey: Liferay.Language.get('others')
		}
	];
};

const groupDeviceData = (data, max) => {
	if (data.length <= max) {
		return data.map(group => ({
			...group,
			label: getDeviceLabel(group.type)
		}));
	}

	const otherData = data.slice(max).reduce(
		(actual, next) => ({
			...actual,
			data: [
				{
					type: Liferay.Language.get('other'),
					views:
						actual.data[0].views +
						next.data.reduce((acc, next) => acc + next.views, 0)
				}
			],
			percentageOfTotal: actual.percentageOfTotal + next.percentageOfTotal
		}),
		{
			data: [{views: 0}],
			percentageOfTotal: 0,
			type: Liferay.Language.get('other')
		}
	);

	return [
		...data.slice(0, max).map(group => ({
			...group,
			label: getDeviceLabel(group.type)
		})),
		{
			...otherData,
			label: Liferay.Language.get('others')
		}
	];
};

/**
 * MAPPER
 * @description Get Devices Mapper
 * @param {function} getMetric
 */
const getDevicesMapper = getMetric => {
	const mapResultToProps = safeResultToProps(result => {
		const metric = getMetric(result);
		const devices = metric.device.map(device => {
			const data = device.metrics.map(({value, valueKey}) => ({
				percentage: getPercentage(value, metric.value),
				type: valueKey,
				views: value
			}));

			return {
				data,
				percentageOfTotal: getPercentage(device.value, metric.value),
				totalViews: device.value,
				type: device.valueKey
			};
		});

		return {
			browsers: groupBrowserData(metric.browser, MAX_BROWSER),
			devices: groupDeviceData(devices, MAX_SYSTEMS),
			total: metric.value
		};
	});

	/**
	 * Map Props to Options
	 * @param {object} param0 props
	 * @param {object} param1 context
	 */
	const mapPropsToOptions = ({
		filters,
		interval,
		rangeSelectors,
		router: {params}
	}) => getVariables({filters, interval, params, rangeSelectors});

	return {
		options: mapPropsToOptions,
		props: mapResultToProps
	};
};

export {getDevicesMapper};
export default getDevicesMapper;
