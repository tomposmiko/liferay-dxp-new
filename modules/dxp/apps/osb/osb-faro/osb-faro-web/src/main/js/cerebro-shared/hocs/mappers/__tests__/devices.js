import {getDevicesMapper} from '../devices';

describe('Shared HOCs Mappers - Devices', () => {
	it('should map devices information', () => {
		const mapper = getDevicesMapper(
			result => result.form.submissionsMetric
		);

		const data = {
			data: {
				form: {
					submissionsMetric: {
						browser: [
							{
								value: 2248.0,
								valueKey: 'Chrome'
							},
							{
								value: 782.0,
								valueKey: 'Firefox'
							},
							{
								value: 137.0,
								valueKey: 'Chrome Mobile'
							},
							{
								value: 136.0,
								valueKey: 'Safari'
							},
							{
								value: 124.0,
								valueKey: 'Mobile Safari'
							},
							{
								value: 109.0,
								valueKey: 'Edge'
							},
							{
								value: 39.0,
								valueKey: 'Unknown'
							},
							{
								value: 20.0,
								valueKey: 'Samsung Browser'
							},
							{
								value: 17.0,
								valueKey: 'Opera Desktop'
							},
							{
								value: 10.0,
								valueKey: 'Chromium Project'
							},
							{
								value: 8.0,
								valueKey: 'Android'
							},
							{
								value: 7.0,
								valueKey: 'Unknown iOS App'
							},
							{
								value: 6.0,
								valueKey: 'Vivaldi'
							},
							{
								value: 5.0,
								valueKey: 'Opera Mini for iOS'
							},
							{
								value: 5.0,
								valueKey: 'Unknown Linux App'
							},
							{
								value: 4.0,
								valueKey: 'Opera Mobile'
							},
							{
								value: 3.0,
								valueKey: 'Yowser for Windows'
							},
							{
								value: 2.0,
								valueKey: 'Firefox for Mobile'
							},
							{
								value: 2.0,
								valueKey: 'Unknown Mac OS X App - WebKit Engine'
							},
							{
								value: 2.0,
								valueKey: 'Waterfox'
							},
							{
								value: 1.0,
								valueKey: 'Chrome for iOS'
							},
							{
								value: 1.0,
								valueKey: 'Edge Mobile'
							},
							{
								value: 1.0,
								valueKey: 'QQ Browser'
							},
							{
								value: 1.0,
								valueKey: 'UBrowser'
							},
							{
								value: 1.0,
								valueKey: 'Yowser for Linux'
							}
						],
						device: [
							{
								metrics: [
									{
										value: 2065.0,
										valueKey: 'Windows'
									},
									{
										value: 995.0,
										valueKey: 'macOS'
									},
									{
										value: 136.0,
										valueKey: 'Linux'
									},
									{
										value: 107.0,
										valueKey: 'Mac OS X'
									},
									{
										value: 53.0,
										valueKey: 'Ubuntu'
									},
									{
										value: 3.0,
										valueKey: 'ChromeOS'
									},
									{
										value: 2.0,
										valueKey: 'Fedora'
									},
									{
										value: 1.0,
										valueKey: 'Unknown'
									}
								],
								value: 3362.0,
								valueKey: 'Desktop'
							},
							{
								metrics: [
									{
										value: 151.0,
										valueKey: 'Android'
									},
									{
										value: 61.0,
										valueKey: 'iOS'
									},
									{
										value: 9.0,
										valueKey: 'Unknown'
									},
									{
										value: 8.0,
										valueKey: 'FreeBSD'
									},
									{
										value: 1.0,
										valueKey: 'Windows Phone'
									}
								],
								value: 230.0,
								valueKey: 'SmartPhone'
							},
							{
								metrics: [
									{
										value: 67.0,
										valueKey: 'iOS'
									},
									{
										value: 12.0,
										valueKey: 'Android'
									}
								],
								value: 79.0,
								valueKey: 'Tablet'
							}
						]
					}
				}
			}
		};

		expect(mapper.props(data)).toMatchSnapshot();
	});

	it('should map devices information in correct order', () => {
		const mapper = getDevicesMapper(
			result => result.form.submissionsMetric
		);

		const data = {
			data: {
				form: {
					submissionsMetric: {
						browser: [
							{
								value: 9.0,
								valueKey: 'Chrome'
							}
						],
						device: [
							{
								metrics: [
									{
										value: 2.0,
										valueKey: 'Windows'
									},
									{
										value: 1.0,
										valueKey: 'macOS'
									}
								],
								value: 3.0,
								valueKey: 'Desktop'
							},
							{
								metrics: [
									{
										value: 4.0,
										valueKey: 'Android'
									}
								],
								value: 4.0,
								valueKey: 'SmartPhone'
							},
							{
								metrics: [
									{
										value: 1.0,
										valueKey: 'Tizen'
									}
								],
								value: 1.0,
								valueKey: 'Tv'
							},
							{
								metrics: [
									{
										value: 1.0,
										valueKey: 'Linux'
									}
								],
								value: 1.0,
								valueKey: 'SmallScreen'
							}
						]
					}
				}
			}
		};

		expect(mapper.props(data)).toMatchSnapshot();
	});
});
