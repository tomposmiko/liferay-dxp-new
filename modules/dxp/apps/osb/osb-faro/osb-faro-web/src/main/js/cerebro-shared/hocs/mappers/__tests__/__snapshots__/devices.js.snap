// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shared HOCs Mappers - Devices should map devices information 1`] = `
Object {
  "browsers": Array [
    Object {
      "value": 2248,
      "valueKey": "Chrome",
    },
    Object {
      "value": 782,
      "valueKey": "Firefox",
    },
    Object {
      "value": 137,
      "valueKey": "Chrome Mobile",
    },
    Object {
      "value": 136,
      "valueKey": "Safari",
    },
    Object {
      "value": 124,
      "valueKey": "Mobile Safari",
    },
    Object {
      "value": 109,
      "valueKey": "Edge",
    },
    Object {
      "value": 39,
      "valueKey": "Unknown",
    },
    Object {
      "value": 20,
      "valueKey": "Samsung Browser",
    },
    Object {
      "value": 76,
      "valueKey": "Others",
    },
  ],
  "devices": Array [
    Object {
      "data": Array [
        Object {
          "percentage": 0,
          "type": "Windows",
          "views": 2065,
        },
        Object {
          "percentage": 0,
          "type": "macOS",
          "views": 995,
        },
        Object {
          "percentage": 0,
          "type": "Linux",
          "views": 136,
        },
        Object {
          "percentage": 0,
          "type": "Mac OS X",
          "views": 107,
        },
        Object {
          "percentage": 0,
          "type": "Ubuntu",
          "views": 53,
        },
        Object {
          "percentage": 0,
          "type": "ChromeOS",
          "views": 3,
        },
        Object {
          "percentage": 0,
          "type": "Fedora",
          "views": 2,
        },
        Object {
          "percentage": 0,
          "type": "Unknown",
          "views": 1,
        },
      ],
      "label": "Desktop",
      "percentageOfTotal": 0,
      "totalViews": 3362,
      "type": "Desktop",
    },
    Object {
      "data": Array [
        Object {
          "percentage": 0,
          "type": "Android",
          "views": 151,
        },
        Object {
          "percentage": 0,
          "type": "iOS",
          "views": 61,
        },
        Object {
          "percentage": 0,
          "type": "Unknown",
          "views": 9,
        },
        Object {
          "percentage": 0,
          "type": "FreeBSD",
          "views": 8,
        },
        Object {
          "percentage": 0,
          "type": "Windows Phone",
          "views": 1,
        },
      ],
      "label": "Phone",
      "percentageOfTotal": 0,
      "totalViews": 230,
      "type": "SmartPhone",
    },
    Object {
      "data": Array [
        Object {
          "percentage": 0,
          "type": "iOS",
          "views": 67,
        },
        Object {
          "percentage": 0,
          "type": "Android",
          "views": 12,
        },
      ],
      "label": "Tablet",
      "percentageOfTotal": 0,
      "totalViews": 79,
      "type": "Tablet",
    },
  ],
  "error": null,
  "loading": false,
  "refetch": undefined,
  "total": undefined,
}
`;

exports[`Shared HOCs Mappers - Devices should map devices information in correct order 1`] = `
Object {
  "browsers": Array [
    Object {
      "value": 9,
      "valueKey": "Chrome",
    },
  ],
  "devices": Array [
    Object {
      "data": Array [
        Object {
          "percentage": 0,
          "type": "Windows",
          "views": 2,
        },
        Object {
          "percentage": 0,
          "type": "macOS",
          "views": 1,
        },
      ],
      "label": "Desktop",
      "percentageOfTotal": 0,
      "totalViews": 3,
      "type": "Desktop",
    },
    Object {
      "data": Array [
        Object {
          "percentage": 0,
          "type": "Android",
          "views": 4,
        },
      ],
      "label": "Phone",
      "percentageOfTotal": 0,
      "totalViews": 4,
      "type": "SmartPhone",
    },
    Object {
      "data": Array [
        Object {
          "percentage": 0,
          "type": "Tizen",
          "views": 1,
        },
      ],
      "label": "TV",
      "percentageOfTotal": 0,
      "totalViews": 1,
      "type": "Tv",
    },
    Object {
      "data": Array [
        Object {
          "type": "Other",
          "views": 1,
        },
      ],
      "label": "Others",
      "percentageOfTotal": 0,
      "type": "Other",
    },
  ],
  "error": null,
  "loading": false,
  "refetch": undefined,
  "total": undefined,
}
`;
